servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/SequentialExecutor.java

```diff
@@ -49,11 +49,21 @@ public interface ExceptionHandler {
 
     private final ExceptionHandler exceptionHandler;
     private final AtomicReference<Cell> tail = new AtomicReference<>();
+    @Nullable
+    private Thread currentDrainingThread;
 
     SequentialExecutor(final ExceptionHandler exceptionHandler) {
         this.exceptionHandler = requireNonNull(exceptionHandler, "exceptionHandler");
     }
 
+    public boolean isCurrentThreadDraining() {
```

Comment:
This is only really necessary so we can make `DefaultLoadBalancer.usedHosts` non-volatile but it feels 'nice'. Feedback welcome on it's utility.